src/test/java/org/opentripplanner/middleware/docs/SwaggerTest.java

```diff
@@ -0,0 +1,393 @@
+                // The reference is typically in the form "#/definitions/TypeName".
+                // Extract the "raw" type (TypeName in example above) from the definition string.
+                String typeName = refNode.asText().substring(DEFINITIONS_REF.length());
+                typeReferences.add(typeName);
+            }
+
+            // List types from the definitions node that don't have a reference from the list above.
+            // Then remove unused types from the definitions node.
+            List<String> typesToRemove = getFieldNames(definitionsNode, type -> !typeReferences.contains(type));
+            definitionsNode.remove(typesToRemove);
+
+            unusedTypeCount = typesToRemove.size();
+        } while(unusedTypeCount > 0);
+    }
+
+    /**
+     * Insert the definition for {@link:AbstractUser}
+     * and add fields (with description) from AbstractUser to OtpUser.
+     * (spark-swagger ignores parent classes when generating swagger.)
+     * TODO: Find a better way to include parent classes.
+     */
+    private void insertAbstractUserRefs(ObjectNode rootNode, JsonNode definitionsTemplate) {
+        ObjectNode definitionsNode = (ObjectNode) rootNode.get("definitions");
+
+        // Insert the AbstractUser type.
+        definitionsNode.set("AbstractUser", definitionsTemplate.get("AbstractUser"));
+
+        ObjectNode abstractUserRefNode = JsonNodeFactory.instance.objectNode();
+        abstractUserRefNode.put($REF_FIELD, DEFINITIONS_REF + "AbstractUser");
+
+        // On the OtpUser type, insert an "allOf" entry to include the ref to AbstractUser.
+        ObjectNode userNode = (ObjectNode) definitionsNode.get("OtpUser");
+        ArrayNode allOfNode = JsonNodeFactory.instance.arrayNode()
+            .add(abstractUserRefNode.deepCopy())
+            .add(userNode);
+        ObjectNode newUserNode = JsonNodeFactory.instance.objectNode()
+            .set("allOf", allOfNode);
+        definitionsNode.set("OtpUser", newUserNode);
+    }
+
+    /**
+     * Verify that version-controlled doc.yaml file is up to date with the latest changes to swagger file auto-generated
+     * from the spark-swagger library (available at {@link #getSwaggerDocs()}.
+     */
+    @Test
+    public void swaggerDocsAreUpToDate() throws IOException, InterruptedException {
+        HttpResponse<String> swaggerResponse = getSwaggerDocs();
+        assertEquals(swaggerResponse.statusCode(), 200);
+        String autoGeneratedSwagger = swaggerResponse.body();
+        JsonNode swaggerJson = YamlUtils.yamlMapper.readTree(autoGeneratedSwagger);
+        String title = swaggerJson.get("info").get("title").asText();
+        LOG.info("Found swagger docs title: {}", title);
+        assertEquals(title, "OTP Middleware");
+        String versionControlledSwagger = Files.readString(versionControlledSwaggerFile);
+        assertEquals(autoGeneratedSwagger, versionControlledSwagger);
+    }
```

Comment:
It seems like this is the only actual test in this file. Most of the code in this file is for actually generating the docs. It seems like putting this in a test is the wrong place for this since it is actual code that performs a task instead of code that verifies a task is done correctly. Therefore, I think every method besides this one should be put into another file in the `src/main/...` directory instead. And the way to initiate documentation generation shouldn't be via running tests, instead it would be better to have some kind of main method that could parse arguments for certain settings of building the docs.